app/Traits/CallsBucketsTrait.php

```diff
@@ -41,7 +41,9 @@ public function authorizeAWS($source)
 
     public function signedUrl(string $file_path, $asset_id, int $transaction)
     {
-        $asset = Asset::where('id', $asset_id)->first();
+        $asset = cacheRemember('asset_signed_url', [$asset_id], now()->addMinute(), function () use ($asset_id) {
```

Comment:
This query can be execute many time for a single request and the `asset_id` is the same. So, I think that it will be a good option to use the cache in this case.